docs/aiohttp-server.asciidoc

```diff
@@ -0,0 +1,124 @@
+[[aiohttp-server-support]]
+=== Aiohttp Server support
+
+Getting Elastic APM set up for your Aiohttp Server project is easy,
+and there are various ways you can tweak it to fit to your needs.
+
+[float]
+[[aiohttp-server-installation]]
+==== Installation
+
+Install the Elastic APM agent using pip:
+
+[source,bash]
+----
+$ pip install elastic-apm
+----
+
+or add `elastic-apm` to your project's `requirements.txt` file.
+
+
+[float]
+[[aiohttp-server-setup]]
+==== Setup
+
+To set up the agent, you need to initialize it with appropriate settings.
+
+The settings are configured either via environment variables,
+the application's settings, or as initialization arguments.
+
+You can find a list of all available settings in the <<configuration, Configuration>> page.
+
+To initialize the agent for your application using environment variables:
+
+[source,python]
+----
+from aiohttp import web
+
+from elasticapm.contrib.aiohttp import ElasticAPM
+
+app = web.Application()
+
+apm = ElasticAPM(app)
+----
+
+To configure the agent using `ELASTIC_APM` in your application's settings:
+
+[source,python]
+----
+from aiohttp import web
+
+from elasticapm.contrib.aiohttp import ElasticAPM
+
+app = web.Application()
+
+app['ELASTIC_APM'] = {
+    'SERVICE_NAME': '<SERVICE-NAME>',
+    'SECRET_TOKEN': '<SECRET-TOKEN>',
+}
+apm = ElasticAPM(app)
+----
+
+[float]
+[[aiohttp-server-usage]]
+==== Usage
+
+Once you have configured the agent,
+it will automatically track transactions and capture uncaught exceptions within Flask.
+
+Capture an arbitrary exception by calling `capture_exception`:
+
+[source,python]
+----
+try:
+    1 / 0
+except ZeroDivisionError:
+    apm.client.capture_exception()
+----
+
+Log a generic message with `capture_message`:
+
+[source,python]
+----
+apm.client.capture_message('hello, world!')
+----
+
+[float]
+[[aiohttp-server-performance-metrics]]
+==== Performance metrics
+
+If you've followed the instructions above, the agent has already installed our middleware.
+This will measure response times, as well as detailed performance data for all supported technologies.
+
+NOTE: due to the fact that `asyncio` drivers are usually separate from their synchronous counterparts,
+specific instrumentation is needed for all drivers.
+The support for asynchronous drivers is currently quite limited.
+
+[float]
+[[aiohttp-server-ignoring-specific-views]]
+===== Ignoring specific routes
```

Comment:
We get this behavior for free, right? This isn't handled in your integration, but rather in the coordinating code, correct? Just taking note for my tornado stuff.

Reply:
Excellent question, had to look it up myself :D. It happens in `Tracer.end_transaction()`, so is framework independent.

https://github.com/elastic/apm-agent-python/blob/ac3b6ffb5893f1b989507eaa671ff8fbcaa53685/elasticapm/traces.py#L525-L549